api/src/com/cloud/network/guru/NetworkGuruAdditionalFunctions.java

```diff
@@ -0,0 +1,12 @@
+package com.cloud.network.guru;
+
+import java.util.Map;
+
+public interface NetworkGuruAdditionalFunctions {
```

Comment:
Wound't it be possible to call these methods inside the `GuestNetworkGuru` methods?
I mean, have `NiciraNvpGuestNetworkGuru` call `finalizeNetworkDesign` and `listAdditionalNicParams` within the methods that are already defined in NetworkGuru?